src/peft/peft_model.py

```diff
@@ -136,32 +143,49 @@ def save_pretrained(self, save_directory: str, safe_serialization: bool = False,
         os.makedirs(save_directory, exist_ok=True)
         self.create_or_update_model_card(save_directory)
 
-        for adapter_name, peft_config in self.peft_config.items():
-            # save only the trainable weights
-            output_state_dict = get_peft_model_state_dict(
-                self, state_dict=kwargs.get("state_dict", None), adapter_name=adapter_name
-            )
-            output_dir = os.path.join(save_directory, adapter_name) if adapter_name != "default" else save_directory
-            os.makedirs(output_dir, exist_ok=True)
+        if selected_adapters is None:
+            selected_adapters = list(self.peft_config.keys())
+        elif selected_adapters is not None:
+            if any(
+                selected_adapter_name not in list(self.peft_config.keys())
+                for selected_adapter_name in selected_adapters
+            ):
+                raise ValueError(
+                    f"You passed an invalid `selected_adapters` arguments, current supported adapter names are"
+                    f" {list(self.peft_config.keys())} - got {selected_adapters}."
+                )
 
-            if safe_serialization:
-                safe_save_file(
-                    output_state_dict, os.path.join(output_dir, SAFETENSORS_WEIGHTS_NAME), metadata={"format": "pt"}
+        for adapter_name, peft_config in self.peft_config.items():
+            if adapter_name in selected_adapters:
```

Comment:
Just a suggestion, this would be a bit more efficient:

```suggestion
        for adapter_name in selected_adapters:
            peft_config = self.peft_config[adapter_name]
```